tfhe/src/shortint/engine/wopbs/mod.rs

```diff
@@ -155,16 +156,27 @@ impl ShortintEngine {
             &mut self.encryption_generator,
         );
 
+        let max_noise_level = MaxNoiseLevel::new(divide_ceil(
+            parameters.carry_modulus.0 * parameters.message_modulus.0 - 1,
+            parameters.message_modulus.0 - 1,
+        ));
+
         let wopbs_server_key = ServerKey {
             key_switching_key: ksk_wopbs_large_to_wopbs_small,
             bootstrapping_key: ShortintBootstrappingKey::Classic(small_bsk),
             message_modulus: parameters.message_modulus,
             carry_modulus: parameters.carry_modulus,
             max_degree: MaxDegree(parameters.message_modulus.0 * parameters.carry_modulus.0 - 1),
+            max_noise_level,
             ciphertext_modulus: parameters.ciphertext_modulus,
             pbs_order: cks.parameters.encryption_key_choice().into(),
         };
 
+        let max_noise_level = MaxNoiseLevel::new(divide_ceil(
```

Comment:
rename to indicate it's the max noise for the shortint server key (non wop)

Reply:
Done